src/data-loader.ts

```
@@ -535,36 +535,93 @@ export function createUniqueHash(data: UsageData): string | null {
+ * - Previously read entire file into memory (slow for large files)
+ * - Now streams and stops after finding timestamp (usually in first line)
+ * - Reduces memory usage and improves performance significantly
  */
 export async function getEarliestTimestamp(filePath: string): Promise<Date | null> {
 	try {
-		const content = await readFile(filePath, 'utf-8');
-		const lines = content.trim().split('\n');
+		// OPTIMIZATION: Only read first few lines instead of entire file
+		// Most JSONL files have timestamps on every line, so first line should suffice
+		// This avoids loading potentially hundreds of MB into memory
+		const { createReadStream } = await import('node:fs');
```

ESLint

[error] Unsafe assignment of an `any` value.

(ts/no-unsafe-assignment)